client/verta/tests/unit_tests/test_endpoint.py

```diff
@@ -2,109 +2,207 @@
-@patch(f'{VERTA_CLASS}.get_access_token', return_value=GET_ACCESS_TOKEN_RESPONSE)
-def test_get_deployed_model_missing_full_url(mock_get_access_token, # pass in patched methods in order
-                                             mock_get_json_by_id,
-                                             mock_get_status,
-                                             mock_endpoint) -> None:
-    """ Verify that the get_deployed_model method returns the correct value for the full_url when it is returned
-       by the get_json_by_id method.  The 'full_url' key was added to the schema of the response 10/2022.
+GET_JSON_BY_ID_RESPONSE_MISSING_URL.pop(
+    "full_url"
+)  # Remove the full url key and value from the dict.
+
+
+@patch(f"{VERTA_CLASS}.get_status", return_value=GET_STATUS_RESPONSE)
+@patch(
+    f"{VERTA_CLASS}._get_json_by_id", return_value=GET_JSON_BY_ID_RESPONSE_MISSING_URL
+)
+@patch(f"{VERTA_CLASS}.get_access_token", return_value=GET_ACCESS_TOKEN_RESPONSE)
+def test_get_deployed_model_missing_full_url(
+    mock_get_access_token,  # pass in patched methods in order
+    mock_get_json_by_id,
+    mock_get_status,
+    mock_endpoint,
+) -> None:
+    """Verify that the get_deployed_model method returns the correct value for the full_url when it is returned
+    by the get_json_by_id method.  The 'full_url' key was added to the schema of the response 10/2022.
     """
     deployed_model = mock_endpoint.get_deployed_model()
-    assert deployed_model.prediction_url == 'https://test_socket/api/v1/predict/test_path'
+    assert (
+        deployed_model.prediction_url == "https://test_socket/api/v1/predict/test_path"
+    )
+
+
+@settings(suppress_health_check=[HealthCheck.function_scoped_fixture])
+@given(
+    kafka_settings=mock_kafka_settings(),
+    kafka_configs_response=mock_kafka_configs_response(),
+)
+def test_kafka_cluster_config_id_default(
+    kafka_settings,
+    kafka_configs_response,
+    mock_endpoint,
+    mock_conn,
+    mocked_responses,
+    mock_registered_model_version,
+) -> None:
+    """Verify that, while updating an endpoint, not including a `cluster_config_id`
+    in the KafkaSettings results in the correct sequence of HTTP requests, including
+    a call to fetch the ID of the current Kafka config by default."""
+    deployment_url = f"{mock_conn.scheme}://{mock_conn.socket}/api/v1/deployment"
+    stages_url = (
+        f"{deployment_url}/workspace/{WORKSPACE_ID}/endpoints/{DEPLOYMENT_ID}/stages"
+    )
+    put_config_url = f"{deployment_url}/stages/654/kafka"
+    builds_url = f"{deployment_url}/workspace/{WORKSPACE_ID}/builds"
+
+    get_configs_url = f"{mock_conn.scheme}://{mock_conn.socket}/api/v1/uac-proxy/system_admin/listKafkaConfiguration"
+
+    with mocked_responses as _responses:
+        # Mock all HTTP requests involved
+        _responses.get(url=stages_url, status=200, json={"id": 654})
+        _responses.get(url=stages_url + "/654", status=200, json={"id": 654})
```

Comment:
I'd really recommend accessing `mock_registered_model_version.id` instead of hard-coding the value into tests. It's not clear in this test which values are kind of arbitrary (like `"cluster_config_id": 987`) versus which are actually tied to fixture values and shouldn't be changed without checking other bits of code.